firefox_ui_tests/functional/locationbar/test_star_in_autocomplete.py

```diff
@@ -19,44 +19,48 @@ class TestStarInAutocomplete(FirefoxTestCase):
     def setUp(self):
         FirefoxTestCase.setUp(self)
 
-        self.clear_history()
+        self.test_urls = [self.marionette.absolute_url('layout/mozilla_grants.html')]
 
         # Location bar suggests 'History and Bookmarks'
         self.prefs.set_pref(self.PREF_LOCATION_BAR_SUGGEST, 0)
 
-    def tearDown(self):
-        FirefoxTestCase.tearDown(self)
-
-        self.prefs.reset_pref(self.PREF_LOCATION_BAR_SUGGEST)
-
-    def clear_history(self):
         with self.marionette.using_context('content'):
             self.marionette.navigate('about:blank')
 
-        self.marionette.execute_script('''
-            let count = gBrowser.sessionHistory.count;
-            gBrowser.sessionHistory.PurgeHistory(count);
-        ''')
+        self.places.remove_all_history()
+
+    def tearDown(self):
+        # Close the autocomplete results
+        try:
+            self.browser.navbar.locationbar.autocomplete_results.close()
+            self.places.restore_default_bookmarks()
+        finally:
+            FirefoxTestCase.tearDown(self)
 
     @skip_under_xvfb
     def test_star_in_autocomplete(self):
         search_string = 'grants'
 
+        def visit_urls():
+            for url in self.test_urls:
+                self.marionette.navigate(url)
+
         with self.marionette.using_context('content'):
-            self.marionette.navigate(self.marionette.absolute_url('layout/mozilla_grants.html'))
+            self.places.wait_for_visited(self.test_urls, visit_urls)
 
+        # TODO: Convert to l10n friendly accessor when menu library is available
         self.browser.menubar.select('Bookmarks', 'Bookmark This Page')
-        done_button = self.marionette.find_element('id', 'editBookmarkPanelDoneButton')
+        # TODO: Replace hard-coded selector with library method when one is available
+        done_button = self.marionette.find_element(By.ID, 'editBookmarkPanelDoneButton')
+        self.wait_for_condition(lambda mn: done_button.is_displayed)
         done_button.click()
 
-        self.clear_history()
+        with self.marionette.using_context('content'):
+            self.marionette.navigate('about:blank')
+
+        self.places.remove_all_history()
 
         locationbar = self.browser.navbar.locationbar
-        # TODO: Replace with visited observer when places module is available
-        # https://bugzilla.mozilla.org/show_bug.cgi?id=1121731
-        # The data for autocomplete does not load immediately and we cannot wait for an observer
-        # until the bug 1121731 is complete
-        time.sleep(4)
         locationbar.clear()
```

Comment:
Maybe you can add some comments to the blocks of code which describe briefly what's done?